### PR TITLE
speed up organization creation by removing some presence validations

### PR DESCRIPTION
**Pull request by dukegreene**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Speed up organization creation by no longer requiring seldom-used fields url, summary, and proof. When values for the fields are provided, we still validate length (and url format) how we used to.

## Related Tickets & Documents

- Related Issue [#19881](https://github.com/forem/forem/issues/19881)
- Closes [#19881](https://github.com/forem/forem/issues/19881)

## QA Instructions, Screenshots, Recordings

Screenshot 2023-08-17 at 6 45 25 PM 

Entering too-long values into the fields....

Screenshot 2023-08-17 at 6 45 43 PM 

...generates errors for length ....

Screenshot 2023-08-17 at 6 52 21 PM 

... and leaving them blank (here's where I got the red asterisks out of the UI too) ....

Screenshot 2023-08-17 at 6 52 31 PM 

... does not generate errors

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

roadrunner saying "beep beep"

### CODE REVIEW

File: app/models/organization.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ class Organization < ApplicationRecord
   validates :text_color_hex, format: COLOR_HEX_REGEXP, allow_blank: true
   validates :twitter_username, length: { maximum: 15 }
   validates :unspent_credits_count, presence: true
-  validates :url, length: { maximum: 200 }, url: { allow_blank: true, no_local: true }
+  validates :url, url: { allow_blank: true, no_local: true }, length: { maximum: 200 }
```
Comment (dukegreene) [inline app/models/organization.rb:61]:
```suggestion
  validates :url, length: { maximum: 200 }, url: { allow_blank: true, no_local: true }
```

**Review by jaw6 — APPROVED** [review]

**Comment by PhilipHow** [comment]
Looks good to me.

**Review by benhalpern — APPROVED** [review]

File: spec/models/organization_spec.rb [hunk 2]
```diff
@@ -407,7 +406,7 @@
 
     context "when the name or profile_image has not changed or the organization has no articles" do
       it "does not call SocialImageWorker.perform_async" do
-        expect(Images::SocialImageWorker).not_to receive(:perform_async)
+        expect(Images::SocialImageWorker).not_to have_received(:perform_async)
```
Comment (lightalloy) [inline spec/models/organization_spec.rb:409]: I suppose it makes sense to `have_received` after `organization.save` (in contrast to `receive` that was previously here)

Comment (dukegreene) [reply spec/models/organization_spec.rb:409]: Yeah I agree, the linter made me do it but it also makes sense and I was also like "sure, I suppose" :-D

**Pull request merged**